components/chainhook-sdk/src/utils/mod.rs

```diff
@@ -403,3 +403,43 @@ pub fn write_file_content_at_path(file_path: &PathBuf, content: &[u8]) -> Result
         .map_err(|e| format!("unable to write file {}\n{}", file_path.display(), e))?;
     Ok(())
 }
+
+#[macro_export]
+macro_rules! try_info {
+    ($a:expr, $tag:expr, $($args:tt)*) => {
+        $a.try_log(|l| slog::info!(l, $tag, $($args)*));
+    };
+    ($a:expr, $tag:expr) => {
+        $a.try_log(|l| slog::info!(l, $tag));
+    };
+}
+
+#[macro_export]
+macro_rules! try_debug {
+    ($a:expr, $tag:expr, $($args:tt)*) => {
+        $a.try_log(|l| slog::debug!(l, $tag, $($args)*));
+    };
+    ($a:expr, $tag:expr) => {
+        $a.try_log(|l| slog::debug!(l, $tag));
+    };
+}
+
+#[macro_export]
+macro_rules! try_warn {
+    ($a:expr, $tag:expr, $($args:tt)*) => {
+        $a.try_log(|l| slog::warn!(l, $tag, $($args)*));
+    };
+    ($a:expr, $tag:expr) => {
+        $a.try_log(|l| slog::warn!(l, $tag));
+    };
+}
+
+#[macro_export]
+macro_rules! try_error {
+    ($a:expr, $tag:expr, $($args:tt)*) => {
+        $a.try_log(|l| slog::error!(l, $tag, $($args)*));
+    };
+    ($a:expr, $tag:expr) => {
+        $a.try_log(|l| slog::error!(l, $tag));
+    };
+}
```

Comment:
There’s a lot of code duplication here. Can we leverage a single core macro to try to reduce the duplication? 🤔 

I am thinking in some like this pseudo code below:

```
#[macro_export]
macro_rules! try_log {
    ... generic macro code here
}

#[macro_export] macro_rules! try_info  { (args) => { try_log!(info,  $(args) }; }
#[macro_export] macro_rules! try_debug { (args) => { try_log!(info,  $(args) }; }
#[macro_export] macro_rules! try_warn  { (args) => { try_log!(info,  $(args) }; }
#[macro_export] macro_rules! try_error { (args) => { try_log!(info,  $(args) }; }

```

Reply:
Hmm I tried this actually but not sure why it doesn't let me refer to a macro defined in the same file from another macro

Reply:
Let's consider this as a future improvement.